lib/commands/exam.js

```diff
@@ -282,7 +282,16 @@ module.exports = TestCommand.extend({
       );
     };
     const browserFailureHandler = function() {
-      if (commands.get('writeExecutionFile')) {
+      // browserFailureHandler is called for disconnect, processError or processExit events.
+      // disconnect and processExit events is fired during global error and successful test runs.
+      // On successful test runs, browserExitHandler should already be called. And is unnecessary
+      // to call it again, so we should return. This is covered by this.finish = true
+      // On global failure cases, it's possible that this.finish is also true. So we must check
+      // any timers set by onProcessExit or onProcessError in testem.
+      // If either timers is set, we should record the failed browser and call browserExitHandler
+      if (this.finished && (!this.onProcessExitTimer && !this.pendingTimer)) {
```

Comment:
Would it be helpful if we have testem source code link where onProcessExitTimer & pendingTimer is set?